### PR TITLE
Refactor function signature map files loading

### PR DESCRIPTION
**Pull request by thg2k**
I apologize for the not-so-great PR, I understand that this class might need a more serious refactoring, but this is just to reduce the copy-paste and make it a bit more readable.

### CODE REVIEW

File: src/Reflection/SignatureMap/FunctionSignatureMapProvider.php [hunk 1]
```diff
@@ -184,80 +184,54 @@ public function getSignatureMap(): array
 		$signatureMap = array_change_key_case($signatureMap, CASE_LOWER);
 
 		if ($this->stricterFunctionMap) {
-			$stricterFunctionMap = require __DIR__ . '/../../../resources/functionMap_bleedingEdge.php';
-			if (!is_array($stricterFunctionMap)) {
-				throw new ShouldNotHappenException('Signature map could not be loaded.');
-			}
-
-			$signatureMap = $this->computeSignatureMap($signatureMap, $stricterFunctionMap);
+			$signatureMap = $this->computeSignatureMapFile($signatureMap, 'functionMap_bleedingEdge.php');
```
Comment (ondrejmirtes) [inline src/Reflection/SignatureMap/FunctionSignatureMapProvider.php]: This makes the file not-clickable in the IDE which is worse experience.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Reflection/SignatureMap/FunctionSignatureMapProvider.php (hunk 1 above)
Comment (thg2k) [reply src/Reflection/SignatureMap/FunctionSignatureMapProvider.php]: I didn't think about that. This can be easily fixed, see follow up

**Comment by thg2k** [comment]
To solve the IDE clickability issue, does it also need the "require" keyword together with the file or the full path is enough?

Also notice I changed require to include, that is to get the more meaningful exception in case the file is missing for whatever reason, instead of just a php error.

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**